### PR TITLE
DD-2723 Updated ModelFieldPointOfInterest to support phone number.

### PR DESCRIPTION
**Pull request by MarkMatute**

### CODE REVIEW

**Comment by MarkMatute** [comment]
HI @tienshiao , to pin point my changes here are the lines:
68 - 70
73 - 81
114
120 - 123
364
402 - 409

File: client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js [hunk 1]
```diff
@@ -114,14 +111,18 @@ angular.module('dashboard.directives.ModelFieldPointsOfInterest', [
...
 				scope.displayedSearchResults = [];     // Data for location list
+				scope.displayOptions = scope.property.display.options;
 				scope.isMapLoading = true;
 				scope.isLoaded = false;
 				scope.placeType = scope.property.display.options.placeType; //Default query value
 				scope.googleApiKey = scope.property.display.options.googleApiKey;
 				scope.googleType = [convertStringToGoogleTypeFormat(scope.placeType)];
+				scope.initalLoad = true;
+				scope.$watch('data.phoneNumber', function(newVal, oldVal) {
+					if (!newVal) scope.data.phoneNumber = oldVal;
```
Comment (tma-isbx) [inline client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js]: Why is this necessary?

File: client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js [hunk 2]
```diff
@@ -65,22 +65,20 @@ angular.module('dashboard.directives.ModelFieldPointsOfInterest', [
            <option value="5" label="5 Miles">5 Miles</option> \
            <option value="10" label="10 Miles">10 Miles</option> \
            <option value="20" label="20 Miles">20 Miles</option> \
-        	 <option value="30" label="30 Miles">30 Miles</option> \
-         </select> \
-        <button class="btn btn-default" ng-click="doSearch()" ng-model="request.query" ng-disabled="disabled">Search</button><span class="search-error" ng-if="searchError">{{searchError}}</span>\
-        </accordion-group>\
-        </accordion> \
-        <div class="map-canvas"id="map_canvas"></div> \
-        <ul class="selected-location" ng-model="displayedSearchResults" > \
```
Comment (tma-isbx) [inline client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js:74]: I think we can probably get rid of/refactor some of the code related to this list.
- displayedSearchResults
- repeatExpression
- updateSelection

File: client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js [hunk 3]
```diff
@@ -359,7 +361,7 @@ angular.module('dashboard.directives.ModelFieldPointsOfInterest', [
 							scope.displayedSearchResults.push(result);
 						}
 					}
-					if (scope.data.placeId) { // pre-select point if exist
+					if (scope.data.placeId && !scope.initalLoad) { // pre-select point if exist
```
Comment (tma-isbx) [inline client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js]: How come we don't want it on the initialLoad?

File: client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js [hunk 4]
```diff
@@ -359,7 +361,7 @@ angular.module('dashboard.directives.ModelFieldPointsOfInterest', [
...
 						perviouslySavedLatLng = new google.maps.LatLng(scope.data.lat, scope.data.lng);
 						scope.getClickedMarker(perviouslySavedLatLng);
```
Comment (tma-isbx) [inline client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js]: Though this doesn't really seem to do anything for me anyways.

File: client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js (hunk 1 above)
Comment (MarkMatute) [reply client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js]: This was added to prevent phonenumber to be null. Removed on the update!

File: client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js (hunk 3 above)
Comment (MarkMatute) [reply client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js]: The reason why we need to skip the initial loading is because we want to display the data from "Our" database rather than the data from the "Maps". Scenarios for this is, if the address was edited by the user, then saved, when the modal was loaded again, data from "Map" will be displayed from the form, and not the data from "Our" database[Edited Data].

File: client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js (hunk 2 above)
Comment (MarkMatute) [reply client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js:74]: repeatExpression, updateSelection was removed.
"displayedSearchResults" is still being used.

Comment (tma-isbx) [reply client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js:74]: I think displayedSearchResults is to support the checkboxes that were removed right? Along with the click handler and listSearchResults().

Comment (MarkMatute) [reply client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js:74]: Yeah you're right! Already removed those code blocks on the recent update. Thanks!

File: client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js [hunk 5]
```diff
@@ -113,15 +109,16 @@ angular.module('dashboard.directives.ModelFieldPointsOfInterest', [
 				scope.boundaries = [];                 // Stored google boundary circles
 				scope.searchResults = [];              // All data recieved from query
 				scope.displayedMarkers = [];           // Markers that match query
-				scope.displayedSearchResults = [];     // Data for location list
+				scope.displayOptions = scope.property.display.options;
```
Comment (tma-isbx) [inline client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js]: Where is this used?

Comment (MarkMatute) [reply client/common/directives/ModelFieldPointsOfInterest/ModelFieldPointsOfInterest.js]: Its not used anymore, ill remove it. Thanks
